frontend/src/employee-frontend/components/messages/MessageContext.tsx

```diff
@@ -98,13 +102,23 @@ const appendMessageAndMoveThreadToTopOfList = (
 
 export const MessageContextProvider = React.memo(
   function MessageContextProvider({ children }: { children: JSX.Element }) {
-    const [selectedAccount, setSelectedAccount] = useState<AccountView>()
+    const { roles } = useContext(UserContext)
+    const messagingEnabled = useMemo(
+      () => requireRole(roles, 'UNIT_SUPERVISOR', 'STAFF'),
+      [roles]
+    )
 
     const [accounts, setAccounts] = useState<Result<MessageAccount[]>>(
       Loading.of()
     )
-    const loadAccounts = useRestApi(getMessagingAccounts, setAccounts)
+    const getAccounts = useRestApi(getMessagingAccounts, setAccounts)
+    const loadAccounts = useDebouncedCallback(getAccounts, 100)
+
+    useEffect(() => {
+      messagingEnabled && loadAccounts()
```

Comment:
I commented about this to @mapehe so I have to comment to you too 😄  I prefer `if (messagingEnabled) loadAccounts()` style when not computing a value but just iffing for side effects.